packages/stateful/actions/components/TransferCw721.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  ArrowRightAltRounded,
+  SubdirectoryArrowRightRounded,
+} from '@mui/icons-material'
+import { useEffect, useMemo, useState } from 'react'
+import { useFormContext } from 'react-hook-form'
+import { useTranslation } from 'react-i18next'
+
+import {
+  AddressInput,
+  ImageEmoji,
+  InputErrorMessage,
+  NftCard,
+  NftSelectionModal,
+} from '@dao-dao/stateless'
+import {
+  ActionComponent,
+  LoadingDataWithError,
+  NftCardInfo,
+} from '@dao-dao/types'
+import { validateAddress, validateRequired } from '@dao-dao/utils'
+
+import { ActionCard } from './ActionCard'
+
+export interface TransferCw721Options {
+  // The set of NFTs that may be transfered as part of this action.
+  options: LoadingDataWithError<NftCardInfo[]>
+  // Information about the NFT currently selected.
+  nftInfo: NftCardInfo | undefined
+}
+
+export const TransferCw721Component: ActionComponent<TransferCw721Options> = ({
+  fieldNamePrefix,
+  onRemove,
+  isCreating,
+  errors,
+  options: { options, nftInfo },
+}) => {
+  const { t } = useTranslation()
+  const { watch, setValue, setError, register, clearErrors } = useFormContext()
+
+  const tokenId = watch(fieldNamePrefix + 'tokenId')
+  const collection = watch(fieldNamePrefix + 'collection')
+
+  const selected = useMemo(
+    () => `${collection}:${tokenId}`,
+    [tokenId, collection]
+  )
```

Comment:
nice. fixed. thanks for the description.

q b/c i find javascript performance interesting: do you think there is anywhere in our codebase that we actually _need_ to use `useMemo` to prevent user-noticeable performance degradation?

Reply:
our codebase has no expensive operations. network requests are the best example of operations we don't want to perform every render, and those happen in `useEffect`s (e.g. TVL/DAO count) and recoil selectors. so we shouldn't need to use `useMemo`s almost at all.

unfortunately, we started out by using `useMemo`s to maintain reference consistency (which is bad practice but we're in too deep now), so a good amount of code expects references not to change and weird loading or flickering states happen otherwise. eventually we should probably fix this, though I doubt users will ever notice.

this is the longest living tech debt, we will see how long it survives